src/mw-utils/services/mw_loading_service_test.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+describe('Loading', function () {
+  var service, rootScope;
+  var $timeout;
+  beforeEach(module('mwUI.Utils'));
+
+  afterEach(function () {
+    service = null;
+    rootScope = null;
+    $timeout = null;
+  });
+
+  beforeEach(inject(function (_$rootScope_, _Loading_, _$timeout_) {
+    rootScope = _$rootScope_;
+    service = _Loading_;
+    $timeout = _$timeout_;
+  }));
+
+  it('can be initialized', function () {
+    expect(service).toBeDefined();
+  });
+
+  it('stores a loading key', function () {
+    service.start('test-key');
+    expect(service.isLoading('test-key')).toBeTruthy();
+    service.done('test-key');
+    expect(service.isLoading('test-key')).toBeFalsy();
+  });
+
+  it('isLoading can handle undefined as key', function () {
+    expect(service.isLoading(undefined)).toBeFalsy();
+  });
+
+  it('isLoading can handle undefined as key and returns true when global loading process is in progress', function () {
+    service.start();
+    expect(service.isLoading()).toBeTruthy();
+  });
+
+  it('isLoading can handle undefined as key and returns false when no global loading process is in progress anymore', function () {
+    service.start();
+    service.done();
+    $timeout.flush();
+
+    expect(service.isLoading()).toBeFalsy();
+  });
+
+  it('isLoading can handle undefined as key and returns true when at least one global loading process is in progress', function () {
+    service.start();
+    service.start();
+
+    service.done();
+
```

Comment:
Add $timeout.flush()... i expect that the test will crash

Reply:
I added a check in afterEach to verify that there is no outstanding $timeout task